snapshots/lcow/lcow.go

```diff
@@ -232,6 +232,17 @@ func (s *snapshotter) Remove(ctx context.Context, key string) error {
 	path := s.getSnapshotDir(id)
 	renamed := s.getSnapshotDir("rm-" + id)
 	if err := os.Rename(path, renamed); err != nil && !os.IsNotExist(err) {
+		log.G(ctx).WithError(err).Errorf("rename failure during snapshot %s cleanup", id)
+		// Sometimes if there are some open handles to the files (especially VHD)
+		// inside the snapshot directory the rename call will return "access
+		// denied" or "file is being used by another process" errors.  Just
+		// returning that error causes the entire snapshot garbage collection
+		// operation to fail. To avoid that we return failed pre-condition error
+		// here so that snapshot garbage collection can continue and can cleanup
+		// other snapshots.
+		if strings.Contains(err.Error(), "being used by another process") || os.IsPermission(err) {
+			return errdefs.ErrFailedPrecondition
```

Comment:
What is the expected remediation when this condition occurs? Is it simply to try the `Remove` later or does it require manual intervention? If it could succeed later, I think it is best not to do an Error log but just wrap the original error message in the returned error, such as `errors.Wrap(errdefs.ErrFailedPrecondition, err.Error())`. The garbage collector will then log it all as a consistent warning.

Reply:
It depends on what is keeping the directory busy. If it is some other process then this will automatically resolve after that process exits (or closes the handle) but it might need manual intervention if this is some other issue. I have updated the code to return wrapped warning.